### ISSUE TITLE
Python API - userMetadata option is silently ignored

### ISSUE THREAD

**Issue by sansbacon** [delta-io/delta#722]
Using Databricks runtime 8.2, with the scala or SQL APIs, I can set userMetdata with a commit. However, with python, the userMetadata option is silently ignored and the value is written as null.

```
df.write.format("delta") \
  .mode("overwrite") \
  .option("userMetadata", "overwritten-for-fixing-incorrect-data") \
  .save("/mnt/delta/events")
```

When I look at the table history after this command, the userMetadata column is still null for the commit.

**Comment by zsxwing** [comment]
@sansbacon I'm not aware of any issue like this. Are you able to reproduce this using OSS Delta? If not, it would be great if you can raise this issue to your Databricks support so that they can help you debug this.

A similar issue we have fixed in Databricks Runtime 8.4 is https://github.com/delta-io/delta/commit/a2722f8b17369a47dd8d23696fc4958f022bb496 but it only affects `saveAsTable`.

**Comment by sansbacon** [comment]
I don't have a setup where I can test using OSS Delta. I will contact Databricks support. Thanks!
